packages/web3-common/src/types.ts

```diff
@@ -0,0 +1,37 @@
+export type JSONRPCId = string | number | null;
+export type JSONRPCResult = string | number | boolean | Record<string, unknown>;
+export type JSONRPCIdentifier = '2.0' | '1.0';
+
+export interface JSONRPCError<T = JSONRPCResult> {
```

Comment:
Can you explain the use of `T = JSONRPCResult`

Reply:
`JSONRPCResult` represents basic probable types, but user may know exact `type` of `data` of the a particular request. So with this approach user can pass the type.